R/DataWrangling.R

```diff
@@ -0,0 +1,166 @@
+.potential_reference_treatments = c(
+  'control',
+  'usual_care',
+  'standard_care',
+  'placebo',
+  'no_contact'
+)
+
+#' Find the expected reference treatment from a vector.
+#' This is done by comparing treatment names to expected reference treatment names.
+#' 
+#' @param treatments vector containing all treatment names
+#' @return Name of the expected reference treatment if one is found, else NULL
+FindExpectedReferenceTreatment <- function(treatments) {
+  expected_reference_treatments <- match(.potential_reference_treatments, tolower(treatments))
+  expected_reference_treatments <- expected_reference_treatments[!is.na(expected_reference_treatments)]
+  if (length(expected_reference_treatments) > 0) {
+    return(treatments[expected_reference_treatments[1]])
+  } else {
+    return(NULL)
+  }
+}
+
+#' Find all of the treatment names in the data, both for long and wide formats.
+#' 
+#' @param data Data frame in which to search for treatment names
+#' @return Vector of all treatment names
+ReplaceTreatmentIds <- function(data, treatent_ids) {
+  if ('T' %in% colnames(data)) {
+    # Long format
+    data$T <- treatent_ids$Number[match(data$T, treatent_ids$Label)]
+  } else {
+    # Wide format
+    for (col in paste0('T.', seq(6))) {
+      if (col %in% colnames(data)) {
+        data[[col]] <- treatent_ids$Number[match(data[[col]], treatent_ids$Label)]
+      } else {
+        break
+      }
+    }
+  }
+  return(data)
+}
+
+#' Add a new column in the data for study IDs, both for long and wide formats.
+#' 
+#' @param data Data frame in which to search for treatment names
+#' @return Vector of all treatment names
+AddStudyIds <- function(data) {
+  # Gather studies and assign IDs
+  study_names <- unique(data$Study)
+  study_df <- data.frame(id = seq(length(study_names)), name = study_names)
+  
+  # Add study IDs to data frame
+  data$StudyID <- study_df$id[match(data$Study, study_df$name)]
+  
+  return(data)
+}
+
+.common_order = c("StudyID", "Study")
+.continuous_specific_order = unlist(
+  lapply(
+    c("", paste0(".", 1:6)),
+    function(x) paste0(c("T", "N", "Mean", "SD"), x)
+  )
+)
+.binary_specific_order = unlist(
+  lapply(
+    c("", paste0(".", 1:6)),
+    function(x) paste0(c("T", "R", "N"), x)
+  )
+)
+
+.continuous_order <- c(.common_order, .continuous_specific_order)
+.binary_order <- c(.common_order, .binary_specific_order)
+
+#' Reorder data frame columns to the correct order, both for long and wide formats.
+#' 
+#' @param data Data frame to reorder
+#' @param outcome_type Type of outcome for which to reorder, either 'Continuous' or 'Binary'
+#' @return Data frame with columns reordered
+ReorderColumns <- function(data, outcome_type) {
+  if (outcome_type == "Continuous") {
+    expected_order <- .continuous_order
+  } else if (outcome_type == "Binary") {
+    expected_order <- .binary_order
+  } else {
+    stop(paste0("Outcome type ", outcome_type, " not recognised. Use either 'Continuous' or 'Binary'"))
```

Comment:
You're right that it couldn't happen in the app. It's more for making the business logic more robust so that if anyone used the code for analysis without the UI, then they'd get more helpful errors.